src/service.js

```diff
@@ -454,9 +450,7 @@ function registerServiceInternal(holder, context, id, opt_ctor, opt_factory) {
     return;
   }
 
-  s.build = () => {
-    return opt_ctor ? new opt_ctor(context) : opt_factory(context);
-  };
+  s.build = () => factory(context);
```

Comment:
We need to capture the context to use for instantiating later. We could alternatively put it on the service object (`s` here).